src/Common/src/TypeSystem/IL/Stubs/TypeSystemThrowingILEmitter.cs

```diff
@@ -61,7 +61,7 @@ public static MethodIL EmitIL(MethodDesc methodThatShouldThrow, TypeSystemExcept
 
             foreach (var arg in exception.Arguments)
             {
-                codeStream.Emit(ILOpcode.ldstr, emitter.NewToken(arg));
+                codeStream.Emit(ILOpcode.ldstr, emitter.NewToken(arg ?? ""));
```

Comment:
This means we're passing more formatting arguments than the exception wants. Are you sure this won't throw at runtime while trying to construct the exception? This might be my fault for not testing it, but this solution only moves the problem to somewhere else.